### PR TITLE
add manual run with bulkop for all orders; update tagging to use GraphQL; update shipping statuses to use GraphQL enums

### PR DESCRIPTION
**Pull request by tekhaus**
When adding a manual mode for all runs, the existing REST calls for create/update were switched to GraphQL to allow for a single order(s) processing loop.

This update assumes that switching the shipping status labels to GraphQL enums won't affect any shops that have this task auto-upgraded (because by definition they haven't modified the task).

These additional statuses are available in GraphQL, but they were not added to keep feature parity with the REST versions:
SUBMITTED      
CANCELED      
FULFILLED      
LABEL_VOIDED      
MARKED_AS_FULFILLED      
NOT_DELIVERED      
PICKED_UP      

Wondering if any of them should be added though. The shipping status link in the task docs (purposely) still links to the REST version, since it provides a good deal more description that the GraphQL docs.

### CODE REVIEW

**Comment by tekhaus** [comment]
[Upgrades for "Auto-tag orders by shipment status"](https://trello.com/c/3nwRo4QO/27-upgrades-for-auto-tag-orders-by-shipment-status)

**Comment by tekhaus** [comment]
Oh, should there be any additional order filters for the bulk op query? Such as only processing "open" orders?

**Comment by isaacbowen** [comment]
Nice and clean. Thank you. :)

> Wondering if any of them should be added though. The shipping status link in the task docs (purposely) still links to the REST version, since it provides a good deal more description that the GraphQL docs.

Oo, interesting. ... If we're switching to GraphQL as the source of truth (which we are doing and which I _want_ to be doing), then I think I want to support everything that's supported there.

I'm okay with replacing the documentation link with https://shopify.dev/docs/admin-api/graphql/reference/shipping-and-fulfillment/fulfillmentdisplaystatus. That's the canonical reference for these values. You could do something like this, with the docs:

```md
For a reference on Shopify's fulfillment statuses, see the [FulfillmentDisplayStatus documentation](https://shopify.dev/docs/admin-api/graphql/reference/shipping-and-fulfillment/fulfillmentdisplaystatus). (More details may be available in the [REST documentation for Fulfillment.shipment_status](https://shopify.dev/docs/admin-api/rest/reference/shipping-and-fulfillment/fulfillment#shipment-status-property-2021-04).)
```

> Oh, should there be any additional order filters for the bulk op query? Such as only processing "open" orders?

Good idea. :) New option: `options.only_tag_orders_matching_this_query_during_bulk_tagging`? The label needs to be super explicit about the query _not_ applying to ongoing fulfillment activity.

File: docs/auto-tag-orders-by-shipment-status/script.liquid [hunk 1]
```diff
@@ -1,28 +1,141 @@
...
+    {% if display_statuses contains shipping_status %}
+      {% unless order.tags contains tag %}
+        {% assign tags_to_add[tags_to_add.size] = tag %}
+      {% endunless %}
+    {% elsif order.tags contains tag %}
+      {% assign tags_to_remove[tags_to_remove.size] = tag %}
+    {% endif %}
+  {% endfor %}
+
+  {% if tags_to_add != empty or tags_to_remove != blank %}
```
Comment (tekhaus) [inline docs/auto-tag-orders-by-shipment-status/script.liquid:112]: Note to self - update this to `tags_to_remove != empty`

**Pull request merged**